### DISCUSSION TITLE
VERCEL ALWAYS FAILS TO INSTALL psycopg2-binary

### DISCUSSION THREAD

**Discussion by bphariharan1301** [vercel/community#2214] [Help]
### Summary

I have a django app which I'm trying to host on vercel. It also has a postrgresql Database hosted on neon. When I try to install the requirements from the requirements.txt file it always throws this error

>  Error: Command failed: pip3.9 install --disable-pip-version-check --target . --upgrade -r /vercel/path0/requirements.txt
   error: subprocess-exited-with-error

which obviously a sub-process error occurring while trying to install psycopg2-binary (version 2.9.5). The error is:

> python setup.py egg_info did not run successfully
Error: pg_config executable not found.
pg_config is required to build psycopg2 from source.  Please add the directory containing pg_config to the $PATH or specify the full executable path with the option:
 python setup.py build_ext --pg-config /path/to/pg_config build or with the pg_config option in 'setup.cfg'

So, would be helpful if you can tell me where I'm going wrong or I'm I missing something. 

I'm also attaching information present in build.sh, settings.py, vercel.json, wsgi.py

I'm also, attaching my repo link for the app which I'm trying to deploy in-case you want to replicate the issue.

Thank you for the help!

### Additional information

```
# build.sh

#!/bin/bash
# Build the project

echo "Building the project"
python3.9 -m pip install -r requirements.txt

echo "Make Migration..."
python3.9 manage.py makemigrations --noinput
python3.9 manage.py migrate --noinput

echo "Collect Static..."
python3.9 manage.py collectstatic --noinput --clear

echo "Done"
```

```
# settings.py
DATABASES = {
    'default': dj_database_url.config(
        default=os.environ.get('DATABASE_URL'), conn_max_age=100
    )
}
```

```
# .env

DATABASE_URL='postgres://<username>:<password>@endpoint-id-123456.us-east-2.aws.neon.tech/<DB_name>?options=project%3Dendpoint-id-123456'
```

```
# vercel.json

{
  "builds": [
    {
      "src":"Portfolio_Website/wsgi.py",
      "use":"@vercel/python",
      "config": {
        "maxLambdaSize": "15mb",
        "runtime":"python3.9"
      }    
    }
  ],
  "routes": [
    {
      "src":"/static/(.*)",
      "dest":"/static/$1"
    },
    {
      "src":"/(.*)",
      "dest":"Portfolio_Website/wsgi.py"
    }
  ]
}
```
```
import os

from django.core.wsgi import get_wsgi_application

os.environ.setdefault('DJANGO_SETTINGS_MODULE', 'Portfolio_Website.settings')

application = get_wsgi_application()
app = application

```

### Example

https://github.com/bphariharan1301/Django-Portfolio

**Comment by 21131f0012** [comment]
did u rectfied this issue.if yes how?
